core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/UnwrapCastInComparison.java

```diff
@@ -171,6 +172,14 @@ public Expression rewriteComparisonExpression(ComparisonExpression node, Void co
             return unwrapCast(expression);
         }
 
+        @Override
+        public Expression rewriteBetweenPredicate(BetweenPredicate node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
```

Comment:
UnwrapCastsInComparison shouldn't be doing this kind of desugaring. Either it needs to handle a BetweenPredicate directly (and produce one in its place), or the desugaring should happen elsewhere so that it only deals with comparison expressions.

Reply:
of course